### DISCUSSION TITLE
ASReview simulation: How to link record_id to the input CSV?

### DISCUSSION THREAD

**Discussion by zhaokoh** [asreview/asreview#1321] [Q&A] [answered]
Hi all,

I am using Python API to access the file after an ASReview simulation, I can get all the required information I need so far (such as time to discovery, etc).

However, I am interested to look at which eligible articles were discovered last, and by using:

` td = time_to_discovery(s) ` and I retrieved say the last 20 items in the list.

According to the time of discovery, these will be the relevant articles that the simulation discovered last.

So, the question is, now, how can I link the record id to the input CSV file (which has the title, abstract, etc)?

Initially, I thought the record_id reflects the row number in the CSV but it does not seem correct. Any suggestion would be much appreciated. Thanks.

**Comment by rohitgarud** [comment, marked as answer]
Hi @zhaokoh, I am also working on some simulations myself. You can try the following code. Replace file paths with your relative paths.

```python
import pandas as pd
from asreview import open_state
from asreview import ASReviewData

dataset = ASReviewData.from_file("tmp_data/api_simulation/data/van_de_Schoot_2017.csv")
df = dataset.df

with open_state("tmp_data/api_example.asreview") as s:
    tds = time_to_discovery(s)
    rids = [rid for rid, td in tds]

    
print(tds)   

rids_last_20 = rids[-20:]
df_last_found = df.loc[rids_last_20, :]
df_last_found 
```

**Reply by Rensvandeschoot** [reply]
Have you seen the [script generator Makita](https://github.com/asreview/asreview-makita)? It includes some helpful [scripts](https://github.com/asreview/asreview-makita#add-and-use-scripts) to obtain the ATD. If there need to be adjustments, please start an issue and/or pull request in that repository. There are 1001 more options to add to Makita, we just started this adventure :-)

**Reply by zhaokoh** [reply]
Thanks @rohitgarud  and @Rensvandeschoot. Oh wow, didn't know Makita exists! Looks like a very powerful tool for simulation studies! Will check it out.

**Reply by zhaokoh** [reply]
@rohitgarud Your code works brilliantly, thanks! I think `ASReviewData.from_file` is what I need. So, it seems like the rows in the CSV files will be reordered after it gets loaded.
